### PR TITLE
Final Milestone:- Add complete react metrics project

### PR DESCRIPTION
**Pull request by hmunish**
I have added following changes:-

- Initiated react app with create-react-app
- Install WebPack bundler
- Install Jest testing library to test the application.
- Install redux toolkit & react-redux packages.
- Add linters config files & install Eslint & Stylelint node packages.
- Add GitHub actions for linters check.
- Install react-test-renderer for snapshot testing.
- Add react-prop-types as a project dependency
- Add async thunk for fetching all currency data
- Add async thunk for fetching conversion rates for a specific currency
- Add extra reducers to update the state with the result of async thunks
- Add event listeners & dispatch actions on click of a specific currency
- Add routes & navbar to navigate through the website
- Add currencyCard, currencyList & currencyDetails JSX components & styles to display on the user interface
- Add redux-mock-store dependency
- Add snapshot testing for components
- Successfully Deployed website using Netlify
- Add professional readme with project description
- Recorded project video description & added the link to readme.md

### CODE REVIEW

File: src/components/navbar.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <nav data-testid="nav-test">
+      <NavLink to="/">
+        <li id="back" />
+      </NavLink>
+      <p>{currencySlice.curCountryCode || 'All country currencies'}</p>
+      <ul>
+        <li id="mic" />
+        <li id="settings" />
+      </ul>
+    </nav>
```
Comment (iambenkis) [inline src/components/navbar.js]: - [x] Good job implementing a very beautiful project using React 👍 . As you required in the guidelines concerning the style, you should follow the [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)) . However, you didn't implement the `hamburger` button 😄  on the left of the navbar. Kindly add it 👍 

### Your design

Capture

### Nelson's design

Capture1

File: src/components/currencies/currencyList.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        <div />
+        <h3>World currencies names</h3>
+      </div>
+      <h1 className="title">Stats by currency</h1>
+      <ul className="countryList">
+        {arr.map((el) => (
+          <CurrencyCard key={el[0]} curCode={el[0]} curName={el[1]} />
+        ))}
+      </ul>
+    </section>
```
Comment (iambenkis) [inline src/components/currencies/currencyList.js]: - [x] Hey, great job with the project 😄 , I admire your effort so far, You've done well. I just wanted to point out something concerning the search. As it's a requirement to add a search functionality so a user could search by currency name, I don't see it working as intended. when a user searches for a currency, it should show up what matches the search  Please, try to add this implementation to meet this part of the project requirement. 👍

### Neslon's design

Capture

### Requirement

Capture

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi @hmunish 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️  Use of react-router
✔️ Follow the design guidelines
✔️ Use Redux for state management

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/App.css [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+#search-bar {
+  float: right;
+  outline: none;
+  border: none;
+  border-radius: 5px;
+  padding: 5px;
+  position: fixed;
+  right: 0;
+  top: 15px;
```
Comment (algerina) [inline src/App.css]: - [ ] You did well implementing the search bar however the position still need some adjustment, I recommend changing the padding to 8px, top of 10 px and right 8px for example.


Screenshot 2023-08-02 at 10 47 08 PM

**Review by algerina — CHANGES_REQUESTED** [review]
Hi @hmunish ,

Great job so far!  You still have some changes to implement.

May I highlight the following: 
- [x] Your project is professionally documented👏  ✔️
- [x] Your App is looking great👏  ✔️



## CHANGES REQUIRED:muscle::clap:


Check the comments under the review.



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 🎉  🎉  🎉  🎉  🎉 
Hi @hmunish, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Good UI design :white_check_mark: 
- [x] Application works as expected :white_check_mark:  
- [x] State management is done using `Redux` :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Pull request has a descriptive title and summary :white_check_mark:  


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**